src/malli/transform.cljc

```diff
@@ -415,6 +415,8 @@
                                                             (-transform-if-valid key-schema)
                                                             (-transform-map-keys))
                                                     (-transform-map-keys m/-keyword->string))))})
+                   (assoc :map {:compile (fn [_ _]
+                                           (-transform-map-keys -string->keyword))})
```

Comment:
I suspect this might need similar `map-of-key-decoders` logic as `map-of` to work in the general case. Alternatively, this might need to be an optional feature like `json-vectors` below.

Reply:
yes, this should not be the default as it blows the perf (your JSON decoder most likely already emits keyword keys) + this will break maps like:

```clojure
[:map
 ["kikka" :string]
 ["kukka" :string]]
```

Reply:
> this will break maps like:

@ikitommi Yeah that's a good point – I've made this transformation schema-aware.

> this should not be the default as it blows the perf (your JSON decoder most likely already emits keyword keys)

True, I've now made this functionality opt-in via `::keywordize-map-keys` option of the transformer.

It's a bit sad that we have to opt in in order to have "decoded value validates against schema" property, but I guess there's no other way (other than creating JSON deserialiser from Malli schemas directly to fuse away this step).